diff --git a/CMakeLists.txt b/CMakeLists.txt
index 911d9f9..5bc80d6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -46,12 +46,9 @@ message(STATUS "Enable C++ trace: ${XGRAMMAR_ENABLE_CPPTRACE}")
 if(MSVC)
   set(CMAKE_CXX_FLAGS "/Wall ${CMAKE_CXX_FLAGS}")
 else()
-  if(NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
-    set(CMAKE_CXX_FLAGS "-O3 ${CMAKE_CXX_FLAGS}")
-  endif()
 
   set(CMAKE_CXX_FLAGS "-Wall -Wextra -Werror -Wno-pedantic -Wno-unused-parameter \
--Woverloaded-virtual -flto=auto ${CMAKE_CXX_FLAGS}"
+-Woverloaded-virtual ${CMAKE_CXX_FLAGS}"
   )
 endif()
 
diff --git a/pyproject.toml b/pyproject.toml
index 11fae7d..bf0b1a9 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -20,8 +20,7 @@ dependencies = [
   "torch>=1.10.0",
   "transformers>=4.38.0",
   "triton; platform_system == 'Linux' and platform_machine == 'x86_64'",
   "mlx-lm; platform_system == 'Darwin' and platform_machine == 'arm64'",
-  "ninja",
 ]
 dynamic = ["version"]
 
